midend/flattenInterfaceStructs.cpp

```diff
@@ -57,21 +82,36 @@ const IR::StructInitializerExpression* StructTypeReplacement::explode(
         }
         vec->push_back(new IR::NamedExpression(f->name, expr));
     }
-    return new IR::StructInitializerExpression(fieldType->name, *vec, false);
+    return new IR::StructInitializerExpression(is_struct ? st->name : ht->name,
+                                               *vec, false);
 }
 
-static const IR::Type_Struct* isNestedStruct(const P4::TypeMap* typeMap, const IR::Type* type) {
+static const IR::Type* isNestedStruct(const P4::TypeMap* typeMap, const IR::Type* type) {
```

Comment:
Type_StructLike

Reply:
> Type_StructLike

Please wait for new code which uses StructLike.